### DISCUSSION TITLE
Setting num_workers  for Torch DataLoader

### DISCUSSION THREAD

**Discussion by ranchy** [acids-ircam/RAVE#102] [General]
Hi, 
I am getting a warning using Google Colab on google drive that I need to set the worker processes to 2. It seems having 8 makes the training slow. I have looked for some kind of config file where I can set num_worker=2. Can someone direct me to where I can set the number of worker processes?   Thanks.

Epoch 11990:  56% 20/36 [00:03<-1:59:56, -3.75it/s, v_num=3]
Validation: 0it [00:00, ?it/s]
Validation:   0% 0/1 [00:00<?, ?it/s]
Validation DataLoader 0:   0% 0/1 [00:00<?, ?it/s]
Epoch 11990: 100% 36/36 [00:07<00:00,  7.04s/it, v_num=3]
Epoch 11991:   0% 0/35 [00:00<00:00, -204457.72it/s, v_num=3]/content/miniconda/lib/python3.9/site-packages/torch/utils/data/dataloader.py:487: UserWarning: This DataLoader will create 8 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(_create_warning_msg(
Epoch 12033:  57% 20/35 [00:04<-1:59:56, -3.61it/s, v_num=3]

**Comment by ranchy** [comment]
I changed BATCH=8 to BATCH=2  in train_rave.py thinking that would do it, however I am still getting the same warning message.

**Reply by AffeZwei** [reply]
If you go to line 125 in train_rave.py, you can change "else 8" to "else 2". 
`num_workers = 0 if os.name == "nt" else 8`

TBH I don't think it makes much difference, at least in my runs. Maybe it depends on your dataset.

**Comment by ranchy** [comment]
A related question, I assume a number of people have run a training on clean_testset_wav, which I am trying now. How many hours does it typically take to train on clean_testset_wav?
thanks
